[Site Admin]: Add a Documentation application story

retrofox opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

~# Branched off from https://github.com/Automattic/wp-calypso/pull/101820~

This PR introduces a new story to the package, aiming to demonstrate how the entire implementation works.

It leverages the components provided by the package for the UI, and uses the routing system to handle UI updates accordingly, and so on. It will be updated as long as the package adds more resources.

Related to #

## Proposed Changes

* [Site Admin]: Add a Documentation application story

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

*

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1) Run the Storybook instance of the Site Admin package
```sh
yarn workspace @automattic/site-admin run storybook:start
```

2) Check the Documentation story


https://github.com/user-attachments/assets/c160dc9f-ef17-4d93-a6a3-69d4c14e71ec


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

retrofox committed:
First approach of the app story implementation

Add component documentation and improve sidebar navigation
Add component documentation and improve sidebar navigation
- Add detailed component descriptions and documentation
- Link to component stories for live examples
- Improve sidebar navigation item labels
- Add HStack for better layout control

ciampo commented:
I'm not sure if using a Storybook example as documentation for the package itself is a good idea — any change to the components will need to be reflected in the examples too. Same for having hardcoded links to the individual stories. Maybe the general example could be a bit closer to a real-world scenario — with a "posts" and "settings" page, with little content each?

I'll try to leave a more in-depth review in the next days if no one else gets to it

retrofox committed:
page: expose component

retrofox committed:
use Page in the App story

retrofox committed:
tweak app story layout

retrofox committed:
change app pages

retrofox committed:
add margin to main area

retrofox committed:
page: rename CSS class names

retrofox committed:
Add PageActions component to story screens
Add action buttons to Pages, Categories and Tags screens with dynamic titles and settings icon

mirka reviewed (commented):
I'm seeing this in the main docs page for some reason 🤔 The story page itself is as expected.

<img width="1081" alt="App example in main docs page" src="https://github.com/user-attachments/assets/a2b4bc47-e966-4cb6-b797-877b3831cc65" />

I do share @ciampo's sentiment that this is _a lot_ of code for a Storybook example — in our experience, this increases maintenance cost, as well as the risk of docs drifting out of date without anybody noticing. This is not necessarily a blocker, and I can defer to your judgement since you probably have a good idea of what kind of documentation and maintenance work is actually necessary in this context. Just raising it as a concern that jumps out.

I would suggest looking into cutting stuff that isn't necessary like the Dashboard content, or reusing code we already have in the individual component stories like the sidebar.

retrofox commented:
<!-- linear:replyTo:2807297193 -->

Thanks for the thoughtful feedback 🙇‍♂️ 

The main goal of this story is to demonstrate how all the components work together in a realistic setup, which isn’t easy to achieve with isolated examples. The amount of code is a side effect of that complexity. Many of these components are interdependent, and that was a core assumption behind this implementation.

Cutting content like the Dashboard would go against that purpose, but I do agree that there might be opportunities to reuse code from existing component stories. I’ll look into that 👍

I also see the increased maintenance cost as a potential advantage. It gives us a way to catch unintended side effects when making changes to individual components, something that’s really hard (if not impossible) to detect when working with limited, isolated stories.

This story is meant less as documentation and more as a realistic playground to stress-test the whole app ecosystem.

I really appreciate the feedback. I’ll keep refining this with that in mind 😄

mirka commented:
> This story is meant less as documentation and more as a realistic playground to stress-test the whole app ecosystem.

Oh this is great actually! I've wanted to add more integration tests like this for custom block editor use cases in Gutenberg.

In that case, I'll just suggest adding CI smoke tests (or more specific behavioral tests) when it feels necessary for QA. Or sharing the fixtures between Storybook and Playwright tests if that makes things simpler. Doesn't have to be now.
